_transcripts/010-rust-1.40.0.md

```diff
@@ -0,0 +1,748 @@
+some useful insights, it's worth reading up on.
+
+__Ben__: So this next one here, borrow check migration warnings have become hard
+errors in Rust 2015, and I think we have spent, I think, maybe, two or three
+previous episodes talking in length about what this means for Rust 2018, and
+Rust 2015, and for the compiler, and for the borrow checker. And so I would
+recommend you just go listen to those episodes. I think 1.39 had some, I think
+1.36, maybe 1.38, possibly. It's just— we've been over this before. If you're on
+Rust 2015 then you need to update your code to accommodate the new borrow
+checker that came last year. And I think it's all we'll say about that in this
+episode.
+
+__Jon__: I think that's reasonable.
+
+__Ben__: We have spent a lot of time. Yeah. Oh, ongoing constification, making
+more standard library functions and methods `const`, usable in constant
+contexts. We really should have some kind of, like, dedicated musical interlude
+intro, because this happens every single release.
+
+__Jon__: It's true.
+
+__Ben__: It's a recurring segment. The `is_power_of_two` function in the
+standard library, which tells you whether or not an integer, is or is not a
+power of two, is now usable in constant contexts.
+
+__Jon__: Is that what it does?
+
+__Ben__: Yeah, yeah. It's kind of subtle, because you couldn't tell.
+
+__Jon__: We should probably have an episode on that function.
+
+__Ben__: Yeah, we could do maybe three or four. And in this case, something more
+interesting, perhaps, that I wanted to talk about is that, last episode we
+mentioned in passing how there was, at that moment, a PR to the Rust compiler to
+enable `if` and `match` expressions to work in constant contexts. And this is a
+big deal, because right now, that's kind of the major thing that's missing.
+Aside from, say, looping. But if `if` and `match` expressions were to work in a
+constant context, it would kind of unblock everything and, kind of like, unleash
+the floodgates for a whole lot more things to become `const fn`s. And so, in
+this case, that PR has since landed in the past six weeks.
+
+And, previously, in order to make certain things `const fn`s in the standard
+library, we had to, kind of, make concessions to readability. And because a lot
+of what was supported previously in `const fn` was mostly bitwise operations, a
+lot of— in order to make things const, a lot of these `const fn`s kind of became
+more unreadable as a result of taking out branches or match statements in order
+to make things work in constant context using only bitwise operations. And that
+was always all of— any time that happened, it was marked with a certain tag in
+the issue tracker called "const fn hack." And so now that `if` and `match`, have
+become— not stable yet, but they are in nightly and usable— the "const fn hacks"
+have been removed. So if you are using nightly Rust, you are in fact using
```

Comment:
Is this `const_fn_hack`?  I can't find any historical record of how this is supposed to be formatted.

Reply:
https://github.com/rust-lang/rust/issues?q=label%3Aconst-hack

Reply:
My instinct is to leave the text alone, unless you'd like the transcripts to allow for corrections.  Or I could put that link in there, to help clarify.

Reply:
I think the link may be useful. Maybe also update the show notes to include it?

Reply:
Done.